src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -329,15 +320,68 @@ void EIRPlantLoopHeatPump::doPhysics(EnergyPlusData &state, Real64 currentLoad)
         return;
     }
 
+    // dispatch to specific physics calculations based on the heat pump type
+    if (this->waterSource) {
+        this->doPhysicsWSHP(state, currentLoad);
+    } else if (this->airSource) {
+        this->doPhysicsASHP(state, currentLoad);
+    }
+}
+
+void EIRPlantLoopHeatPump::doPhysicsWSHP(EnergyPlusData &state, Real64 currentLoad)
+{
+
+    // add free cooling at some point, compressor is off during free cooling, temp limits restrict free cooling range
+
+    Real64 availableCapacity = this->referenceCapacity;
+    Real64 partLoadRatio = 0.0;
+
+    this->calcAvailableCapacity(state, currentLoad, availableCapacity, partLoadRatio);
+    this->setPartLoadAndCyclingRatio(state, partLoadRatio);
+
+    // evaluate the actual current operating load side heat transfer rate
+    this->calcLoadSideHeatTransfer(state, availableCapacity);
+
+    // no defrost calculation for WSHP
+    // calculate power usage from EIR curves
+    this->calcPowerUsage(state);
+
+    // evaluate the source side heat transfer rate
+    this->calcSourceSideHeatTransferWSHP(state);
+}
```

Comment:
The WSHP do physics new function `doPhysicsWSHP()`, which is a wrapper for all water source calculations. No defrost calculation call here.